confgenerator/config.go

```diff
@@ -1295,6 +1298,29 @@ func validateComponentKeys[V any](components map[string]V, refs []string, subage
 	return nil
 }
 
+// Pipelines that export prometheus metrics are not allowed to have Ops Agent processors
+func validateNoCustomGMPProcessors(receivers metricsReceiverMap, receiverIDs, processorIDs []string, subagent string, ctx context.Context) error {
+	for _, ID := range receiverIDs {
+		receiver, ok := receivers[ID]
+		if !ok {
+			continue // Some receivers might not be used.
```

Comment:
Actually reading this again - this looks incorrect. If `!ok`, then it looks like a receiver is used in the pipeline but there is no such receiver in the map? 

Does another validator catch this error already or must we also return an error here?